packages/config/src/validate/validations.js

```diff
@@ -199,6 +207,22 @@ const POST_NORMALIZE_VALIDATIONS = [
       functions: { [prevPath[1]]: { node_bundler: bundlers[0] } },
     }),
   },
+  {
+    property: 'functions.*.directory',
+    check: (value, key, prevPath) => prevPath[1] === FUNCTIONS_CONFIG_WILDCARD_ALL,
+    message: 'must be defined on the top-level functions object.',
```

Comment:
We might want to be more explicit on the message. I did not directly understand what "top-level functions object" meant. From an end user perspective, would `[functions.*]` be clearer?